src/core/lib/event_engine/iomgr_engine/wakeup_fd_posix.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/event_engine/iomgr_engine/wakeup_fd_posix.h"
+
+#include <functional>
+#include <memory>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include "src/core/lib/iomgr/port.h"
+
+namespace grpc_event_engine {
+namespace iomgr_engine {
+
+#ifdef GRPC_POSIX_WAKEUP_FD
+
+namespace {
+
+absl::Mutex g_mu;
```

Comment:
we can't have types with non-trivial destructors at static scope... and I don't think we need any of this machinery anyway.

Instead look at my suggestion in wakeup_fd_posix_default.cc.

Reply:
Removed this file and moved its functions to wakeup_fd_posix_default.cc